main/dsp.c

```diff
@@ -429,7 +429,7 @@ struct ast_dsp {
 	int faxmode;
 	int freqmode;
 	int dtmf_began;
-	int display_inband_dtmf_warning;
+	unsigned int display_inband_dtmf_warning:1;
```

Comment:
Unrelated change that snuck in? If not can you explain the rationale for this change?

Reply:
It is related; I reviewed all the code where this is used and set, and it only needs to be a bitfield, not an int.
I considered renaming the variable too for clarity but guess I didn't do that.

Reply:
But you aren't changing any code that reads or sets this structure member, so it's not clear to me what the benefit is. Does it change the generated code? Due to alignment concerns I don't think it would change the size of the struct either?

Reply:
It's more semantic. You can look at the struct and know that it's a boolean variable, which was not conveyed before. I think it helps with code clarity to do this, regardless of space savings, since it serves as documentation.

If more bitfields are added in the future, then it would actually reduce memory usage.

Reply:
> If more bitfields are added in the future, then it would actually reduce memory usage.

I don't think that is correct, but I will defer to your expertise on this.

Reply:
I don't like change for the sake of change. If that time ever comes, then it can be changed at that point. Right now it is premature to touch it or care about it. Just changing the log message is fine.

Reply:
It wasn't really for a future hypothetical performance benefit, it was more for making the code more self-documenting, since that variable is already a misnomer. I've changed it back but added a documentation note about this.

Reply:
Looks like `busymaybe` could be a bitfield too. I would add the same comment there so people aren't confused.